extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -413,20 +413,30 @@ export class AmpAnalytics extends AMP.BaseElement {
    * @param {string} template The template to expand.
    * @param {!JSONObject} The object to use for variable value lookups.
    * @param {!Object} event Object with details about the event.
+   * @param {number} opt_iterations Number of recursive expansions to perform.
+   *    Defaults to 2 substitutions.
    * @return {string} The expanded string.
    * @private
    */
-  expandTemplate_(template, trigger, event) {
+  expandTemplate_(template, trigger, event, opt_iterations) {
+    opt_iterations = opt_iterations === undefined ? 2 : opt_iterations;
+    if (opt_iterations < 0) {
+      return '';
```

Comment:
I do return 12${3}. The thing is that after the call to `this.expandTemplate_`, I encode the value of the variable (need to encode the variable before putting it in URL). The encode messes up with the {}. Since this is an unusual case, I am okay with keeping it as it is. Will also add a warning to log this case.